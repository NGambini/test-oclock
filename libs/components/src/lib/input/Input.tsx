import React, { useRef } from 'react';
import MaterialIcon from '@material/react-material-icon';

import * as styles from './input.module.less';
import { Ripple } from '../utils/ripple/Ripple';
import { reference } from '@popperjs/core';

export type InputProps = {
  value: string;
  placeholder?: string;
  icon?: string;
  onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
};

export const Input = (props: InputProps) => {
  const inputEl = useRef<HTMLInputElement>(null);

  return (
    <div
      data-cy="input"
      className={styles['input-container']} // wrapper is required for Ripple
      onClick={() => inputEl.current.focus()} // forward click down to input
    >
      <Ripple />
      {props.icon && (
        <MaterialIcon className={styles['input-icon']} icon={props.icon} />
      )}
      <input
        ref={inputEl}
        placeholder={props.placeholder}
        onChange={props.onChange}
        className={styles['input']}
        value={props.value}
      />
    </div>
  );
};
