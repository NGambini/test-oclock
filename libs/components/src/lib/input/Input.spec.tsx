import React from 'react';
import { render } from '@testing-library/react';

import { Input } from './Input';

describe('Input', () => {
  it('renders correctly', () => {
    const { baseElement: input } = render(
      <Input icon="account_balance" placeholder="Placeholder" value="" />
    );
    expect(input).toMatchSnapshot();
  });
});
