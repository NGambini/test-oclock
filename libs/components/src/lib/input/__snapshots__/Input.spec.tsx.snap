// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly 1`] = `
<body>
  <div>
    <div
      data-cy="input"
    >
      <div
        class="rippleContainer"
      />
      <i
        class="material-icons"
      >
        account_balance
      </i>
      <input
        placeholder="Placeholder"
        value=""
      />
    </div>
  </div>
</body>
`;
